lib/client.js

```diff
@@ -282,7 +282,9 @@ class Client extends Dispatcher {
         handler = new RedirectHandler(this, maxRedirections, opts, handler)
       }
 
-      const request = new Request(opts, handler)
+      const origin = opts.origin || this[kUrl].origin
+
+      const request = new Request({ origin, ...opts }, handler)
```

Comment:
I would prefer avoiding to recreating this object. This is a hot path. Maybe add a new positional parameter to request?

Reply:
yep I was thinking about this as well, would putting it first would make sense?